### PR TITLE
✅  New decimal-aware price system - Part 2: Fix application tests

### PR DESCRIPTION
**Pull request by JuaniRios**
Main info on #260 

- This PR focuses on fixing the application tests

### CODE REVIEW

File: pallets/funding/src/mock.rs [hunk 1]
```diff
@@ -70,6 +69,7 @@ pub const fn free_deposit() -> Balance {
 use crate::traits::ProvideAssetPrice;
 use frame_support::traits::{Everything, OriginTrait};
 use frame_system::RawOrigin as SystemRawOrigin;
+use polimec_common::USD_DECIMALS;
 use polkadot_parachain_primitives::primitives::Sibling;
 use sp_runtime::traits::{ConvertBack, Get, TryConvert};
 use xcm_builder::{EnsureXcmOrigin, FixedWeightBounds, ParentIsPreset, SiblingParachainConvertsVia};
```
Comment (vstam1) [inline pallets/funding/src/mock.rs:75]: Could we change the file order to:
1. Imports
2. Constants
3. Types
4. Structs

File: pallets/funding/src/tests/mod.rs [hunk 2]
```diff
@@ -15,15 +15,18 @@ use frame_support::{
...
 use sp_arithmetic::{traits::Zero, Percent, Perquintill};
 use sp_runtime::{BuildStorage, TokenError};
 use sp_std::{cell::RefCell, marker::PhantomData};
 use std::iter::zip;
 type MockInstantiator =
 	Instantiator<TestRuntime, <TestRuntime as crate::Config>::AllPalletsWithoutSystem, RuntimeEvent>;
-const US_DOLLAR: u128 = 1_0_000_000_000;
+const USD_UNIT: u128 = 10u128.pow(USD_DECIMALS as u32);
+const USDT_UNIT: u128 = USD_UNIT;
+const ASSET_UNIT: u128 = 10_u128.pow(10u32);
```
Comment (vstam1) [inline pallets/funding/src/tests/mod.rs:28]: Is this `ASSET_UNIT` representing PLMC? If so, would we rename it to PLMC or import PLMC const from some other place? If not, could we use the `ASSET_DECIMALS` here in the pow from line 31?

**Review by vstam1 — COMMENTED** [review]
Looks good, couple of comments.

File: pallets/funding/src/tests/mod.rs (hunk 2 above)
Comment (JuaniRios) [reply pallets/funding/src/tests/mod.rs:28]: It refers to CTs. So lets rename it to CT_UNIT, and put a different decimal count than PLMC

File: pallets/funding/src/mock.rs [hunk 3]
```diff
@@ -20,45 +20,46 @@
...
+use xcm_executor::traits::XcmAssetTransfers;
 
-type Block = frame_system::mocking::MockBlock<TestRuntime>;
+pub const PLMC: Balance = 10u128.pow(PLMC_DECIMALS as u32);
+pub const MILLI_PLMC: Balance = PLMC / 10u128.pow(3);
+pub const MICRO_PLMC: Balance = PLMC / 10u128.pow(6);
+pub const EXISTENTIAL_DEPOSIT: Balance = 10 * MILLI_PLMC;
+pub const USD_UNIT: Balance = 10u128.pow(USD_DECIMALS as u32);
 
+pub type Block = frame_system::mocking::MockBlock<TestRuntime>;
```
Comment (vstam1) [inline pallets/funding/src/mock.rs:53]: Where do we need this Block for? Is it just a move?

**Review by vstam1 — APPROVED** [review]
Lgtm

**Review by lrazovic — APPROVED** [review]

File: pallets/funding/src/mock.rs (hunk 3 above)
Comment (JuaniRios) [reply pallets/funding/src/mock.rs:53]: Yes it moved and I added the pub keyword

**Comment by JuaniRios** [comment]
### Merge activity

* **Apr 30, 9:52 AM EDT**: @JuaniRios started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Polimec/polimec-node/262).

**Pull request merged**